pkg/dashboard/assets/js/namespace_list.js

```diff
@@ -0,0 +1,37 @@
+$(function () {
+    var $namespaceCards = $("#namespaceCards .card")
+    var $searchBox = $("#search")
+
+    // filters namespace cards containing the search text
+    function filterNamespaceCards() {
+        var searchText = $searchBox.val()
+        $namespaceCards.filter(function() {
+            var cardNamespace = $("strong", $(this)).text()
+            var cardContainsSearchText = cardNamespace.toLowerCase().indexOf(searchText.toLowerCase()) > -1
+            if(cardContainsSearchText) {
+                $(this).show()
+            } else {
+                $(this).hide()
+            }
+        })
+    }
+
+    // clicks on the "view" button for the first visible namespace card
+    function viewDashboardForFirstNamespace() {
+        $namespaceCards.filter("div:not([style='display: none;'])").first().find("a[data-name='view'")[0].click()
+    }
+
+    // filter namespace cards on every key typed
+    $searchBox.on("keyup", filterNamespaceCards)
+
+    // visit the first filtered namespace card's view/dashboard page by pressing enter
+    $searchBox.on("keypress", function(e) {
+        var key = e.which;
+        if(key == 13) { // enter
+            viewDashboardForFirstNamespace()
+        }
+    })
```

Comment:
I'm not really a huge fan of this functionality as it is now

There's no indication in the UI of this being how that works. When I type in a search field, I hit enter by default to do the search. This just jumps me somewhere with no warning or indication.

Reply:
I'm not sure I'm seeing the same behaviour that you're seeing. When I type in a Namespace I see the list filtering and then when I hit enter the top result is selected and the dashboard navigates to that Namespace's summary page. Alternatively when I use the mouse to click on a Namespace in the list the dashboard also navigates to that Namespace's summary page.

I'm not a UI designer though, so any suggestions are willingly accepted :)

Reply:
That's exactly the behavior I'm seeing, and it's very unexpected. When I hit enter, I expect to see search results, not be taken to the top result. 

There's really nothing to tell me what just happened or that it is going to happen when I hit enter.

Reply:
We can drop the "visit first namespace on enter" behaviour entirely if you'd like. Then `<enter>` would effectively do nothing and would instead just rely on the filtering. It would be like an illusion of enter.